sdk/search/search-documents/src/synonymMapHelper.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { SynonymMap } from "./serviceModels";
+import { promisify } from "util";
+import * as fs from "fs";
+const readFileAsync = promisify(fs.readFile);
+
+/**
+ * Helper method to create a SynonymMap object. This is a NodeJS only method.
+ *
+ * @param name - Name of the SynonymMap.
+ * @param filePath - Path of the file that contains the Synonyms (seperated by new lines)
+ * @returns SynonymMap object
+ */
+export async function createSynonymMapFromFile(
+  name: string,
+  filePath: string
+): Promise<SynonymMap> {
+  const synonyms: string[] = (await readFileAsync(filePath, "utf-8")).replace("\r", "").split("\n");
```

Comment:
From MDN's page on [replace()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace):

> If pattern is a string, only the first occurrence will be replaced.

We can't rely on [replaceAll ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) since it requires Node 15

To work around this, perhaps something like this?

```suggestion
  const synonyms: string[] = (await readFileAsync(filePath, "utf-8")).replace(/\r/g, "").split("\n");
```